sql/wsrep_server_service.cc

```diff
@@ -302,3 +302,62 @@ int Wsrep_server_service::wait_committing_transactions(int timeout)
 void Wsrep_server_service::debug_sync(const char*)
 {
 }
+
+int Wsrep_server_service::do_crypt(void**                ctx,
+                                   wsrep::const_buffer&  key,
+                                   const char            (*iv)[32],
+                                   wsrep::const_buffer&  input,
+                                   void*                 output,
+                                   bool                  encrypt,
+                                   bool                  last)
+{
+
+  const void* deserialized_key_ptr;
+  size_t deserialized_key_size;
+  unsigned int key_version;
+  if(wsrep_key_deserialize(key.data(), key.size(), deserialized_key_ptr,
+                           deserialized_key_size, key_version))
+  {
+     throw wsrep::runtime_error("Failed wsrep_key_deserialize()");
```

Comment:
Where is this exception handled ?

Reply:
Function is called from encryption_cb, implemented in wsrep-lib and try ... catch block is there to catch runtime_errors.